tests/src/core/testqgsgml.cpp

```diff
@@ -1273,5 +1276,72 @@ void TestQgsGML::testSameTypeameAsGeomName()
   delete features[0].first;
 }
 
+void TestQgsGML::testUnknownEncoding_data()
+{
+  QTest::addColumn<QString>( "xmlHeader" );
+  QTest::addColumn<QByteArray>( "encoding" );
+
+  QTest::newRow( "simple quote" ) << QStringLiteral( "<?xml version='1.0' encoding='ISO-8859-15'?>" ) << QByteArrayLiteral( "ISO-8859-15" );
+  QTest::newRow( "double quote" ) << QStringLiteral( "<?xml version='1.0' encoding=\"ISO-8859-15\"?>" ) << QByteArrayLiteral( "ISO-8859-15" );
+  QTest::newRow( "UTF-8" ) << QStringLiteral( "<?xml version='1.0' encoding=\"UTF-8\"?>" ) << QByteArrayLiteral( "UTF-8" );
+  QTest::newRow( "No header" ) << QString() << QByteArrayLiteral( "UTF-8" );
```

Comment:
it could be interesting to add a test for an invalid/unhandled encoding, to check robustness

Reply:
Done